src/imagecapturecore.cs

```diff
@@ -0,0 +1,901 @@
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+//
+// imagecapturecore.cs: Bindings for the ImageCaptureCore API
+//
+using System;
+using AppKit;
+using Foundation;
+using ObjCRuntime;
+using CoreImage;
+using CoreGraphics;
+using CoreAnimation;
+
+namespace ImageCaptureCore {
+
+	interface IICDeviceDelegate {}
+
+	[Static]
+	interface ICDeviceCapabilities {
+		[Field ("ICDeviceCanEjectOrDisconnect")]
+		NSString DeviceCanEjectOrDisconnect { get; }
+
+		[Field ("ICCameraDeviceCanTakePicture")]
+		NSString CameraDeviceCanTakePicture { get; }
+
+		[Field ("ICCameraDeviceCanTakePictureUsingShutterReleaseOnCamera")]
+		NSString CameraDeviceCanTakePictureUsingShutterReleaseOnCamera { get; }
+
+		[Field ("ICCameraDeviceCanDeleteOneFile")]
+		NSString CameraDeviceCanDeleteOneFile { get; }
+
+		[Field ("ICCameraDeviceCanDeleteAllFiles")]
+		NSString CameraDeviceCanDeleteAllFiles { get; }
+
+		[Field ("ICCameraDeviceCanSyncClock")]
+		NSString CameraDeviceCanSyncClock { get; }
+
+		[Field ("ICCameraDeviceCanReceiveFile")]
+		NSString CameraDeviceCanReceiveFile { get; }
+
+		[Field ("ICCameraDeviceCanAcceptPTPCommands")]
+		NSString CameraDeviceCanAcceptPtpCommands { get; }
+	}
+
+	[Static]
+	interface ICScannerStatus {
+		[Field ("ICScannerStatusWarmingUp")]
+		NSString WarmingUp { get; }
+
+		[Field ("ICScannerStatusWarmUpDone")]
+		NSString WarmUpDone { get; }
+
+		[Field ("ICScannerStatusRequestsOverviewScan")]
+		NSString RequestsOverviewScan { get; }
+	}
+
+	[Static]
+	interface ICDeviceLocationDescriptions {
+		[Field ("ICDeviceLocationDescriptionUSB")]
+		NSString Usb { get; }
+
+		[Field ("ICDeviceLocationDescriptionFireWire")]
+		NSString FireWire { get; }
+
+		[Field ("ICDeviceLocationDescriptionBluetooth")]
+		NSString Bluetooth { get; }
+
+		[Field ("ICDeviceLocationDescriptionMassStorage")]
+		NSString MassStorage { get; }
+	}
+
+	[Static]
+	interface ICCameraDownloadOptionKeys {
+		[Field ("ICDownloadsDirectoryURL")]
+		NSString DownloadsDirectoryUrl { get; }
+
+		[Field ("ICSaveAsFilename")]
+		NSString SaveAsFilename { get; }
+
+		[Field ("ICSavedFilename")]
+		NSString SavedFilename { get; }
+
+		[Field ("ICSavedAncillaryFiles")]
+		NSString SavedAncillaryFiles { get; }
+
+		[Field ("ICOverwrite")]
+		NSString Overwrite { get; }
+
+		[Field ("ICDeleteAfterSuccessfulDownload")]
+		NSString DeleteAfterSuccessfulDownload { get; }
+
+		[Field ("ICDownloadSidecarFiles")]
+		NSString DownloadSidecarFiles { get; }
+	}
+
+	[Static]
+	interface ICStatusNotificationKeys { // Keys into DidReceiveStatusInformation dictionary 
+		[Field ("ICStatusNotificationKey")]
+		NSString NotificationKey { get; }
+
+		[Field ("ICStatusCodeKey")]
+		NSString CodeKey { get; }
+
+		[Field ("ICLocalizedStatusNotificationKey")]
+		NSString LocalizedNotificationKey { get; }
```

Comment:
Coud we create a strong dictionary for this? That would make the API nicer for the users.

Reply:
Would not be very useful since the way the API uses this is using the NSString key to obtain a value from an NSDictionary that comes as a parameter from a **protocol** member method, so having a smart enum would add an additional `.GetConstant ()` (`ICStatusNotificationKeys.NotificationKey.GetConstant ()`) that is not easily discoverable and unnecessary.

```csharp
[Export ("device:didReceiveStatusInformation:")]
public void DidReceiveStatusInformation (ICDevice device, NSDictionary<NSString, NSObject> status)
{
	var state = status[ICStatusNotificationKeys.NotificationKey] as NSString;
	//...
}
```